### ISSUE TITLE
Consider accepting --features="" as a no-op like Cargo does

### ISSUE THREAD

**Issue by HadrienG2** [obi1kenobi/cargo-semver-checks#565]
### Describe your use case

When a crate of mine has features, the associated CI commonly uses a feature matrix like this...

```yaml
    strategy:
      matrix:
        features:
          - '""'
          - 'std'

    # ... later used in a job step like this:
    
      - name: Type-check the program
        run: cargo check --all-targets --features=${{ matrix.features }}
```

Cargo is thankfully fine with me specifying `--features=""` as a no-op here, which is very convenient as it allows me not to duplicate CI job steps for the "no extra features case" with respect to the "with extra features" case.

Unfortunately, cargo semver-checks is not so forgiving, and running `cargo semver-checks --features=""` results in this sort of errors:

```text
     Parsing hwlocality v1.0.0 (current)
Error: `cargo metadata` exited with an error: error: failed to select a version for `hwlocality`.
    ... required by package `rustdoc v0.0.0 (/home/hadrien/Bureau/RustModernization/hwlocality/target/semver-checks/local-hwlocality-1_0_0-e48bd6b743e86099)`
versions that meet the requirements `*` are: 1.0.0

the package `rustdoc` depends on `hwlocality`, with features: `` but `hwlocality` does not have these features.


failed to select a version for `hwlocality` which could resolve this conflict
```

### Describe the solution you'd like

I'd like cargo semver-checks to accept `--features=""` as a no-op like most cargo commands do, keeping my CI workflow simple and clean.

### Alternatives, if applicable

Do nothing. Duplicating the cargo semver-checks CI job step is annoying, but not the end of the world.

**Comment by obi1kenobi** [comment]
Interesting! I've never seen a workload structured like this, but it seems reasonable and it's also good to stay consistent with cargo too.

I'd happily merge a PR with an implementation and a test for this, if you're open to making one? I'm a bit swamped with life things at the moment so I'm not sure when I'd get a chance to get to this otherwise 😅
